msal/managed_identity.py

```diff
@@ -136,6 +136,10 @@ class ManagedIdentityClient(object):
 
     .. note::
 
+        By setting the environment variable ``MSAL_MANAGED_IDENTITY_ENDPOINT`` you override the
+        default identity URL used. This is useful during local development where it may be
+        desirable to utilise the credentials assigned to an actual instance via SSH tunnelling.
+
```

Comment:
Your original wording sounded like this new MSAL_MANAGED_IDENTITY_ENDPOINT behavior would be applicable to all different managed identity (MI) variations. But that is not the case, for good reasons: (1) we want to minimize the reliance of a MSAL-invented env var; (2) most of other MIs already has their canonical `IDENTITY_ENDPOINT` anyway so they won't need a new one.

```suggestion
    .. admonition:: Special case when your local development wants to use a managed identity on Azure VM.

        By setting the environment variable ``MSAL_MANAGED_IDENTITY_ENDPOINT``
        you override the default identity URL used in MSAL's Azure VM managed identity
        code path.

        This is useful during local development where it may be desirable to 
        utilise the credentials assigned to an actual VM instance via SSH tunnelling.

        For example, if you create your SSH tunnel this way (assuming your VM is on x.x.x.x)::

            ssh -L 8000:169.254.169.254:80 x.x.x.x

        Then your code could run locally using::

            export IMDS_ENDPOINT=http://localhost:8000/metadata/identity/oauth2/token; python your_script.py

    .. note::
```

Reply:
Manually merged in with some minor word smithing on the example.

Reply:
Nice. I have updated our doc staging side to reflect that. Doesn't the [new purple box here](https://msal-python.readthedocs.io/en/docs-staging/#msal.ManagedIdentityClient) look good? :-)

Reply:
It does look the biz, thanks!